src/main/java/com/taosdata/jdbc/utils/Utils.java

```diff
@@ -230,4 +234,31 @@ private static boolean isNanoSecFormat(String timestampStr) {
         }
         return true;
     }
+
+    /**
+     * Some of the SQLs sent by other popular frameworks or tools like Spark, contains syntax that cannot be parsed by
+     * the TDengine client. Thus, some simple parsers/filters are intentionally added in this JDBC implementation in
+     * order to process those supported SQLs.
+     */
+    public static String preprocessSql(String rawSql) {
+        //For processing some of Spark SQLs
+        // SELECT * FROM db.tb WHERE 1=0
+        rawSql = rawSql.replaceAll("WHERE 1=0", "WHERE _c0=1");
+        rawSql = rawSql.replaceAll("WHERE 1=2", "WHERE _c0=1");
```

Comment:
_c0 is the first timestamp column. 1 is a valid value if the users intend to store very old data (around the Unix epoch). So the statement has a small chance to fail.

Reply:
suggest to use "WHERE _c0 is null" to replace

Reply:
thanks